corehq/apps/reports/standard/deployments.py

```diff
@@ -243,9 +295,26 @@ def total_records(self):
 
     @property
     def rows(self):
-        users = self.user_query().run()
-        self._total_records = users.total
-        return self.process_rows(users.hits)
+        if self.warehouse:
+            mobile_user_and_group_slugs = set(
+                # Cater for old ReportConfigs
+                self.request.GET.getlist('location_restricted_mobile_worker') +
+                self.request.GET.getlist(ExpandedMobileWorkerFilter.slug)
+            )
+            users = ExpandedMobileWorkerFilter.user_es_query(
+                self.domain,
+                mobile_user_and_group_slugs,
+                self.request.couch_user,
+            ).values_list('_id', flat=True)
+            sort_clause = self.get_sql_sort()
+            rows = ApplicationStatusFact.objects.filter(user_id__in=users).order_by(sort_clause)
```

Comment:
doesn't app status fact have the user dim id now? looks like this is using couch user ids to filter

also how many user ids could be in this list? IN queries usually don't perform well at scale

Reply:
I think this could end up being quite a lot of users, thousands if not more. I had thought of performance concerns related to this, but our filters return a list of user_ids and I did not know how else to structure the query. If there is an alternative query that would be more performant, I would love to rewrite that part.

Reply:
This is a problem with a lot of reports - filters give us a small set of directly selected user IDs, group ids, location ids, and user types.  Currently we turn that into a potentially very large set of user IDs for reporting, and do a query like this.  Other options (way out of the scope of this PR) could be:

* Store more user meta directly in forms, cases, `ApplicationStatusFact`s or whatever your rows are, such as user type and the user's groups and locations (although then we'd have to keep that up-to-date)
* The same as the above, but do a relational lookup like a SQL join.  Obviously that's not possible with a couch backend, and I have no idea if that'd work with postgres sharding.

Not sure if there's an easy solution to this problem, unfortunately.

Reply:
i agree it's probably out of scope of this PR, but the best way to make this perform well would be to insert all the user ids you want to filter on into an indexed temporary table and then join that table to your result set. the process would be something like:

- create a temporary table with one column, user_id, that is indexed
- insert all the user ids into the temporary table
- include in the query a join from ApplicationStatusFact to UserDim, and a join from UserDim to the temporary table
- drop the temporary table after producing the results

Reply:
@esoergel  in many ways the warehouse already does that first bullet. It takes a few joins but in a single query we could grab all app status records for a certain location or group. It seems likely that would be a significant improvement. It would also let us drop the ES query for user ids entirely.

@gcapalbo do you have a sense of whether an `IN` query on say 3 locations (keeping in mind the 2 extra joins that would require) would be significantly faster than ~1000 user ids. I imagine if only a single location was selected which i assume is the dominant usage, it would definitely be more performant that way. I think what I am trying to ask is how `IN` performance degrades. Is it strongly related to the size of the tables or the number of elements you are matching against. This can certainly solve one of those problems, but not the total data size.

I think thats a pretty easy change and long term something we want anyway to eliminate the ES dependency of this (and later other warehouse) reports, so I am willing to make that change now rather than later if it looks like a performance win.